### PR TITLE
Multi label soft : debug + other version of distance correlation

### PR DESCRIPTION
**Pull request by fantes**
in caffelib.cc: read the correct output layer
in supervised output connector : lighter distance correlation + eucll now discards -1 

for testing 
curl -X PUT "http://localhost:8080/services/test_multi" -d '{"model": {"templates": "../templates/caffe/", "repository": "/home/infantes/test/"}, "type": "supervised", "description": "imagenet classifier", "parameters": {"input": {"db": false, "height": 224, "connector": "image", "width": 224, "bw": false, "multi_label": true}, "mllib": {"finetuning": true, "rotate": false, "template": "resnet_50", "gpuid": [0], "db": false, "distort": {"all_effects": true, "prob": 0.01}, "noise": {"all_effects": true, "prob": 0.01}, "weights": "ResNet-50-model.caffemodel", "nclasses": 9, "mirror": true, "gpu": true, "regression": true}, "output": {}}, "mllib": "caffe"}'

for training:
curl -X PUT "http://localhost:8080/train" -d '{"async": true, "data": ["/home/infantes/test/part_shuf_train.txt", "/home/infantes/test/part_shuf_test.txt"], "parameters": {"input": {"db": false, "shuffle": true}, "mllib": {"gpu": true, "net": {"test_batch_size": 4, "batch_size": 16}, "solver": {"test_initialization": true, "base_lr": 0.001, "snapshot": 10000, "iterations": 300000, "test_interval": 100, "iter_size": 2, "solver_type": "SGD"}, "gpuid": [0], "resume": false}, "output": {"measure": ["acc", "eucll"]}}, "service": "test_multi"}'

### CODE REVIEW

**Comment by beniz** [comment]
This has been merged manually.
